contracts/evoting/evoting.go

```diff
@@ -825,7 +882,64 @@ func (e evotingCommand) manageAdminForm(snap store.Snapshot, step execution.Step
 		return xerrors.Errorf("failed to marshal Form : %v", err)
 	}
 
-	err = snap.Set(formID, formBuf)
+	err = snap.Set(formIDBuf, formBuf)
+	if err != nil {
+		return xerrors.Errorf("failed to set value: %v", err)
+	}
+
+	return nil
+}
+
+// initializeAdminList initialize an AdminList on the blockchain. It is called the first time that
+// we attempt to add an admin.
+func initializeAdminList(snap store.Snapshot, initialAdmin int, ctx serde.Context) error {
+	h := sha256.New()
+	h.Write([]byte(AdminListId))
+	formIDBuf := h.Sum(nil)
+
+	adminList := types.AdminList{
+		AdminList: []int{initialAdmin},
+	}
+
+	formBuf, err := adminList.Serialize(ctx)
+	if err != nil {
+		return xerrors.Errorf("failed to marshal Admin Form : %v", err)
+	}
+
+	err = snap.Set(formIDBuf, formBuf)
+	if err != nil {
+		return xerrors.Errorf("failed to set value: %v", err)
+	}
+
+	// Update the form metadata store
+
+	formsMetadataBuf, err := snap.Get([]byte(FormsMetadataKey))
+	if err != nil {
+		return xerrors.Errorf("failed to get key '%s': %v", formsMetadataBuf, err)
+	}
+
+	formsMetadata := &types.FormsMetadata{
+		FormsIDs: types.FormIDs{},
+	}
+
+	if len(formsMetadataBuf) != 0 {
+		err := json.Unmarshal(formsMetadataBuf, formsMetadata)
+		if err != nil {
+			return xerrors.Errorf("failed to unmarshal FormsMetadata: %v", err)
+		}
+	}
+
+	err = formsMetadata.FormsIDs.Add(hex.EncodeToString(formIDBuf))
+	if err != nil {
+		return xerrors.Errorf("couldn't add new form: %v", err)
+	}
+
+	formMetadataJSON, err := json.Marshal(formsMetadata)
+	if err != nil {
+		return xerrors.Errorf("failed to marshal FormsMetadata: %v", err)
+	}
+
+	err = snap.Set([]byte(FormsMetadataKey), formMetadataJSON)
```

Comment:
this seems to me to be code that's probably used elsewhere too - if that's the case, maybe put it in a function that's called?